actions/lib/purge_remote_folder.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/python
+
+import argparse
+import datetime
+import json
+import logging
+import os
+import shutil
+import socket
+
+
+class PurgeRemoteFolders(object):
+
+    def __init__(self, biotank_archive_folder, age_in_days, dryrun):
+        self.biotank_archive_folder = str(biotank_archive_folder)
+        self.age_in_days = int(age_in_days)
+        self.dryrun = bool(dryrun)
+
+        hostname = socket.gethostname()
+        self.directory = self.sanitize_path(
+            self.biotank_archive_folder,
+            os.path.join(
+                "/data",
+                hostname))
+        self.logfile = os.path.join(self.directory, "KEEP_removed_archives.log")
+
+    @staticmethod
+    def sanitize_path(dirty_path, directory):
+        if os.path.isabs(dirty_path):
+            raise ValueError(
+                "{p} is an absolute path which is not allowed at this point".format(
+                    p=dirty_path))
+        # fully resolve path to make sure that no symlinks or upward references are left and cause surprises
+        clean_path = os.path.realpath(
+            os.path.join(directory, dirty_path))
+        # ensure that the resulting path is below the directory
+        if not clean_path.startswith(directory):
+            raise ValueError(
+                "{p} appears to resolve to a path outside of {d} which is not allowed at this point".format(
+                    p=dirty_path, d=directory))
+        return clean_path
+
+    def get_files_and_folders(self):
+        older_than = datetime.date.today() - datetime.timedelta(days=self.age_in_days)
+
+        def _path_should_be_purged(p):
+            return datetime.date.fromtimestamp(
```

Comment:
This could be broken up into multiple variables, e.g.
```
dir_timestamp = ...
days_since_change = fromtimestamp(...)
days_since_change < older_than ...
```
That would make this easier to parse for the reader.

Reply:
ok, good point